torch/_refs/__init__.py

```diff
@@ -4267,15 +4267,13 @@ def lerp(start: Tensor, end: Tensor, weight: Union[Tensor, NumberType]):
         )
     assert isinstance(weight, Tensor)  # mypy
     # We implement it this way for numerical stability. We assume (in the stability optimisation)
-    # that 0 <= weight <= 1. We take the abs to deal with complex numbers
+    # that -1 <= weight <= 1. We take the abs to deal with complex numbers
     # We want to perform operations near zero, which is where floating points are most precise
-    # thus, we perform the following optimisation:
-    # If weight.abs() >= 0.5:
-    #    return (1 - weight) * (start - end) + end
-    mask = weight.abs() >= 0.5
-    coeff = torch.where(mask, weight - 1, weight)
-    base = torch.where(mask, end, start)
-    return coeff * (end - start) + base
+    return torch.where(
+        weight.abs() >= 0.5,
+        start + (end - start) * weight,
+        end + (start - end) * (1 - weight),
+    )
```

Comment:
Have you benchmarked this? There is one fewer where but 2x the number of mathematical operations.

Reply:
That's fair. Actually what this hints is that I can actually optimise the previous PR. I'll drop this one.

Reply:
Nevermind, this optimisation for lerp, given that a few of these arguments are scalars, does not make sense. You get 5 ops + 1 where vs 3 ops + 2 where. But the where implementation with 2 wheres requires creating potentially large tensors from scalars, which is not great. Leaving it as it is and dropping this PR.